src/func.c

```diff
@@ -1222,6 +1223,49 @@ void FuncDeclaration::semantic(Scope *sc)
 
 void FuncDeclaration::semantic2(Scope *sc)
 {
+    //printf("FuncDeclaration::semantic2 [%s] fd0 = %s %s\n", this->loc.toChars(), this->toChars(), this->type->toChars());
+    if (overnext && !errors)
+    {
+        OutBuffer buf1;
+        mangleToFuncSignature(&buf1, this);
+        const char *s1 = buf1.peekString();
+
+        OutBuffer buf2;
+        for (Dsymbol *s = overnext; s; )
+        {
+            if (FuncDeclaration *fd = s->isFuncDeclaration())
+            {
+                s = fd->overnext;
+                if (fd->errors)
+                    continue;
+
+                // Don't have to check conflict between declaration and definition.
+                if ((fbody != NULL) != (fd->fbody != NULL))
+                    continue;
+
+                buf2.reset();
+                mangleToFuncSignature(&buf2, fd);
+                const char *s2 = buf2.peekString();
+
+                //printf("+%s\n\ts1 = %s\n\ts2 = %s @ [%s]\n", toChars(), s1, s2, fd->loc.toChars());
+                if (strcmp(s1, s2) == 0)
+                {
+                    TypeFunction *tf2 = (TypeFunction *)fd->type;
+                    fd->error("%s conflicts with at %s",
+                            parametersTypeToChars(tf2->parameters, tf2->varargs), loc.toChars());
+                    fd->type = Type::terror;
+                    fd->errors = true;
+                }
+            }
+            else if (AliasDeclaration *ad = s->isAliasDeclaration())
+            {
+                // todo
+                s = ad->overnext;
```

Comment:
I suspect much of this function is duplicating what is in overloadApply(), but not handling all the cases.